test/recipes/03-test_fipsinstall.t

```diff
@@ -0,0 +1,40 @@
+#! /usr/bin/env perl
+# Copyright 2019 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+
+use strict;
+use warnings;
+
+use File::Spec;
+use File::Copy;
+use OpenSSL::Glob;
+use OpenSSL::Test qw/:DEFAULT srctop_file/;
+use OpenSSL::Test::Utils;
+
+setup("test_fipsinstall");
+
+plan skip_all => "Test disabled in this configuration"
+    if $^O eq 'MSWin32';
+
+plan tests => 2;
+
+my $infile = srctop_file("providers", "fips.so");
```

Comment:
Should be `bldtop_file`

Also, the name won't necessarily be "fips.so", so I suggest using available perl modules to get the local name, which would be `bldtop_file('providers', platform->dso('fips'))`, and then you shouldn't need to suppress MSWin32.

You get the `platform` package like this:

``` perl
use lib srctop_dir('Configurations');
use lib bldtop_dir('.');
use platform;
```

Reply:
Cheers - not likely I would have figured that out any time soon :)

Reply:
Added a few negative tests also

Reply:
It's easy to forget when you're knee deep in Unix land. I get regular reminders from my VMS builds